controller/control.go

```diff
@@ -1244,25 +1244,17 @@ func (c *Controller) IsReplicaRW(replicaInController *types.Replica) error {
 }
 
 // DeleteSnapshot ...
-func (c *Controller) DeleteSnapshot(replicas []types.Replica) error {
+func (c *Controller) DeleteSnapshot(snapshot string, replicas []types.Replica) error {
 	var err error
 
-	ops := make(map[string][]replica.PrepareRemoveAction)
 	for _, r := range replicas {
 		replica := r // pin it
-		ops[replica.Address], err = c.prepareRemoveSnapshot(&replica, c.SnapshotName)
+		_, err = c.prepareRemoveSnapshot(&replica, snapshot)
 		if err != nil {
 			return err
 		}
 	}
 
-	for _, rep := range replicas {
```

Comment:
don't understand why this is removed

Reply:
as per PR message, deleting snapshots is left to snapshot cleanup go routine.
But, PrepareRemoveDisk is returning nil, nil when removed is true.

Reply:
When a snapshot delete request is received, prepareRemoveDisk is being called which marks the disk as removed.
Once this disk is marked as removed, this is taken up by InternalSnapshotDeletion thread to delete the snapshot.